### PR TITLE
[DCOS-53141] Jobs File Based Secrets Dependencies

### PR DESCRIPTION
**Pull request by brandonc**
## Overview

Some type additions and clarifying renames that will be needed for Jobs File Based Secrets.

Should not change any behavior.

## Important

Hold merge until companion PR is ready

### CODE REVIEW

File: plugins/jobs/src/js/components/form/helpers/JobFormData.ts [hunk 1]
```diff
@@ -57,13 +57,20 @@ export enum Container {
 export interface JobSpec {
   cmdOnly: boolean;
   container?: Container | null;
-  job: JobFormData;
+  job: JobSpecData;
   schedule?: JobSchedule;
 }
 
 export type EnvModel = Array<[string, string]>;
 export type SecretModel = Array<[string, string, string]>;
```
Comment (natmegs) [inline plugins/jobs/src/js/components/form/helpers/JobFormData.ts]: Can we delete this now?

Comment (brandonc) [reply plugins/jobs/src/js/components/form/helpers/JobFormData.ts]: Removed it in a follow up commit

File: plugins/jobs/src/js/components/form/helpers/JobFormData.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export interface JobRun<Env, Secrets> {
   secrets?: Secrets;
 }
 
-export type JobFormData = Job<ArrayLabels, EnvModel, SecretModel>;
+export type JobSpecData = Job<ArrayLabels, EnvModel, JobSecretExposure>;
```
Comment (natmegs) [inline plugins/jobs/src/js/components/form/helpers/JobFormData.ts]: I think we will need `JobSecretExposure[]` here

Comment (brandonc) [reply plugins/jobs/src/js/components/form/helpers/JobFormData.ts]: 😭  of course

**Pull request merged**